llvm/lib/Target/AArch64/AArch64Processors.td

```diff
@@ -718,12 +718,16 @@ def ProcessorFeatures {
   list<SubtargetFeature> AppleA13 = [HasV8_4aOps, FeatureCrypto, FeatureFPARMv8,
                                      FeatureNEON, FeaturePerfMon, FeatureFullFP16,
                                      FeatureFP16FML, FeatureSHA3];
+  // Apple A14 and M1 chips are based on Armv8.5-a but without BTI. Since there
```

Comment:
I don't think we need to emphasize this here or in `aarch64-target-features.c`;  it's just v8.4a with some bits from .5, and there are other instances of that.  This wasn't always explained clearly in the ARM ARM extension list, but the more recent issues make this pattern explicit (e.g., looking at DDI 0487K.a: A2.1.1, or the start of A2.2.5)

Reply:
Is there a problem with emphasising it? The main motivation is to avoid people repeating the mistake that I made, i.e. seeing 8.4 here and 8.5 in publicly available documentation and "correcting" it.

Reply:
Well, on the flipside, having one specific quirk called out somewhere and no other quirk called out elsewhere would only instill misplaced confidence in anyone who would want to update these cpu definitions.

Reply:
No, because this isn't documentation for these cores, nor should it be.  There is no documentation beyond the optimization guide mentioned above publicly available today.

Reply:
As you point out, the only publicly available documentation is the optimization guide, and lists these cores as "8.5 without BTI". Here in the tablegen file it is written as 8.4 plus all of the features from 8.5 which are supported. A comment explaining the discrepancy seems reasonable to me. It also alerts people that if any new backend feature is introduced that models something in 8.5, they need to manually add it to the list for these cores.